site/content/docs/02-template-syntax.md

```diff
@@ -0,0 +1,592 @@
+	<p>Something went wrong: {error.message}</p>
+{/await}
+```
+
+---
+
+The `catch` block can be omitted if no error is possible, and the initial block can be omitted if you don't care about the pending state.
+
+```html
+{#await promise then value}
+	<p>The value is {value}</p>
+{/await}
+```
+
+
+### DOM events
+
+* `on:eventname={handler}`
+* `on:eventname|modifiers={handler}`
+
+---
+
+Use the `on:` directive to listen to DOM events.
+
+```html
+<script>
+	let count = 0;
+
+	function handleClick(event) {
+		count += 1;
+	}
+</script>
+
+<button on:click={handleClick}>
+	count: {count}
+</button>
+```
+
+---
+
+Handlers can be declared inline with no performance penalty. As with attributes, directive values may be quoted for the sake of syntax highlighters.
+
+```html
+<button on:click="{() => count += 1}">
+	count: {count}
+</button>
+```
+
+---
+
+Add *modifiers* to DOM events with the `|` character.
+
+The following modifiers are available:
+
+* `preventDefault` — calls `event.preventDefault()` before running the handler
+* `stopPropagation` — calls `event.stopPropagation()`, preventing the event reaching the next element
+* `passive` — improves scrolling performance on touch/wheel events (Svelte will add it automatically where it's safe to do so)
+* `capture` — fires the handler during the *capture* phase instead of the *bubbling* phase
+* `once` — remove the handler after the first time it runs
+
+Modifiers can be chained together, e.g. `on:click|once|capture={...}`.
+
+```html
+<form on:submit|preventDefault={handleSubmit}>
+	<!-- the `submit` event's default is prevented,
+	     so the page won't reload -->
+</form>
+```
+
+---
+
+If the `on:` directive is used without a value, the component will *forward* the event, meaning that a consumer of the component can listen for it.
+
+```html
+<button on:click>
+	The component itself will emit the click event
+</button>
+```
+
+
+### Component events
+
+* `on:eventname={handler}`
+
+---
+
+Components can emit events using [createEventDispatcher](#docs/createEventDispatcher), or by forwarding DOM events. Listening for component events looks the same as listening for DOM events:
+
+```html
+<SomeComponent on:whatever={handler}/>
+```
+
+
+
+### Element bindings
+
+* `bind:property={value}`
+
+---
+
+Data ordinarily flows down, from parent to child. The `bind:` directive allows data to flow the other way, from child to parent. Most bindings are specific to particular elements.
+
+The simplest bindings reflect the value of a property, such as `input.value`.
+
+```html
+<input bind:value={name}>
+<textarea bind:value={text}></textarea>
+
+<input type="checkbox" bind:checked={yes}>
+```
+
+---
+
+If the name matches the value, you can use a shorthand.
+
+```html
+<!-- These are equivalent -->
+<input bind:value={value}>
+<input bind:value>
+```
+
+---
+
+Numeric input values are coerced; even though `input.value` is a string as far as the DOM is concerned, Svelte will treat it as a number.
```

Comment:
Maybe some comment here about the bound value being `undefined` is an empty string or other invalid number is entered.

Reply:
Ah I forgot about `undefined`. Fixed. It's not possible to enter anything non-numeric into an input with `type="number"` so we don't need to worry about those cases

Reply:
In Chrome yeah, but in Firefox, you are able to enter non-numeric text into `type='number'`.